src/fmt/humantime.rs

```diff
@@ -99,13 +95,40 @@ impl fmt::Debug for Timestamp {
 
 impl fmt::Display for Timestamp {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let formatter = match self.precision {
-            TimestampPrecision::Seconds => format_rfc3339_seconds,
-            TimestampPrecision::Millis => format_rfc3339_millis,
-            TimestampPrecision::Micros => format_rfc3339_micros,
-            TimestampPrecision::Nanos => format_rfc3339_nanos,
+        let Ok(ts) = jiff::Timestamp::try_from(self.time) else {
+            return Err(fmt::Error);
+        };
+
+        match self.precision {
+            TimestampPrecision::Seconds => write!(f, "{ts:.0}"),
+            TimestampPrecision::Millis => write!(f, "{ts:.3}"),
+            TimestampPrecision::Micros => write!(f, "{ts:.6}"),
+            TimestampPrecision::Nanos => write!(f, "{ts:.9}"),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::Timestamp;
+    use crate::TimestampPrecision;
+
+    #[test]
+    fn test_display_timestamp() {
+        let mut ts = Timestamp {
+            time: std::time::SystemTime::UNIX_EPOCH,
+            precision: TimestampPrecision::Nanos,
         };
 
-        formatter(self.time).fmt(f)
+        assert_eq!("1970-01-01T00:00:00.000000000Z", format!("{ts}"));
+
+        ts.precision = TimestampPrecision::Micros;
+        assert_eq!("1970-01-01T00:00:00.000000Z", format!("{ts}"));
+
+        ts.precision = TimestampPrecision::Millis;
+        assert_eq!("1970-01-01T00:00:00.000Z", format!("{ts}"));
+
+        ts.precision = TimestampPrecision::Seconds;
+        assert_eq!("1970-01-01T00:00:00Z", format!("{ts}"));
     }
 }
```

Comment:
Can you add the tests (passing) in their own commit before the implementation change?  That way, this commit shows there is no behavior change (or shows what it is and we can sign off on it)

Reply:
@epage Good idea. Implemented!